src/pybind/mgr/dashboard/controllers/auth.py

```diff
@@ -1,40 +1,30 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import
 
-import time
-
 import cherrypy
 
 from . import ApiController, RESTController
 from .. import logger
 from ..exceptions import DashboardException
-from ..services.auth import AuthManager
-from ..tools import Session
+from ..services.auth import AuthManager, JwtManager
 
 
 @ApiController('/auth', secure=False)
 class Auth(RESTController):
     """
-    Provide login and logout actions.
-
-    Supported config-keys:
-
-      | KEY             | DEFAULT | DESCR                                     |
-      ------------------------------------------------------------------------|
-      | session-expire  | 1200    | Session will expire after <expires>       |
-      |                           | seconds without activity                  |
+    Provide authenticates and returns JWT token.
     """
 
-    def create(self, username, password, stay_signed_in=False):
-        now = time.time()
+    def create(self, username, password):
         user_perms = AuthManager.authenticate(username, password)
         if user_perms is not None:
-            cherrypy.session.regenerate()
-            cherrypy.session[Session.USERNAME] = username
-            cherrypy.session[Session.TS] = now
-            cherrypy.session[Session.EXPIRE_AT_BROWSER_CLOSE] = not stay_signed_in
             logger.debug('Login successful')
+            token = JwtManager.gen_token(username)
+            token = token.decode('utf-8')
+            logger.debug("JWT Token: %s", token)
```

Comment:
This looks like a security vulnerability.

IMO, access tokens are considered security critical data that should not be logged.

Reply:
Indeed, storing login credentials in logs could be considered a potential security vulnerability. But this seems to be a log entry that only appears when the log level is set to "debug", which is not the production default level, or is it? We may have a similar issue in #24475 then.

Reply:
Ok, if this is acceptable, everything else LGTM.